### ISSUE TITLE
Docs are failing to build due to calling diagram.drop

### ISSUE THREAD

**Issue by danyeaw** [gaphor/gaphor#3622]
### Describe the bug
Docs are currently failing due to a drop command that is referenced in `gaphor-services.md`.

```{code-cell} ipython3
from gaphor.core.modeling import Diagram, StyleSheet
from gaphor.diagram.drop import drop

element_factory.create(StyleSheet)
diagram = element_factory.create(Diagram)

for element in element_factory.lselect():
    drop(element, diagram, x=0, y=0)
```

### To Reproduce
1. Build the docs
1. See error

```
 Running Sphinx v8.1.2
loading translations [en]... done
Settings schema not found and settings won’t be saved. Run `gaphor install-schemas`.
making output directory... done
Gaphor models: {'connect': 'connect.gaphor', 'design-principles': 'design-principles.gaphor', 'stereotypes': 'stereotypes.gaphor', 'example': '../examples/sequence-diagram.gaphor', 'coffee-machine': '../examples/coffee-machine.gaphor', 'sysml-template': '../gaphor/templates/sysml.gaphor', 'style-sheets': 'style-sheet-examples.gaphor', 'core': '../models/Core.gaphor', 'uml': '../models/UML.gaphor', 'sysml': '../models/SysML.gaphor', 'raaml': '../models/RAAML.gaphor', 'c4model': '../models/C4Model.gaphor'}
loading intersphinx inventory 'gnome' from https://amolenaar.pages.gitlab.gnome.org/pygobject-docs/objects.inv ...
loading intersphinx inventory 'pygobject' from https://pygobject.gnome.org/objects.inv ...
loading intersphinx inventory 'gaphas' from https://gaphas.readthedocs.io/en/stable/objects.inv ...
loading intersphinx inventory 'python' from https://docs.python.org/3/objects.inv ...
myst v4.0.0: MdParserConfig(commonmark_only=False, gfm_only=False, enable_extensions=set(), disable_syntax=[], all_links_external=False, links_external_new_tab=False, url_schemes=('http', 'https', 'mailto', 'ftp'), ref_domains=None, fence_as_directive=set(), number_code_blocks=[], title_to_header=False, heading_anchors=3, heading_slug_func=None, html_meta={}, footnote_sort=True, footnote_transition=True, words_per_minute=200, substitutions={}, linkify_fuzzy_links=True, dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', enable_checkboxes=False, suppress_warnings=[], highlight_code_blocks=True)
myst-nb v1.1.2: NbParserConfig(custom_formats={}, metadata_key='mystnb', cell_metadata_key='mystnb', kernel_rgx_aliases={}, eval_name_regex='^[a-zA-Z_][a-zA-Z0-9_]*$', execution_mode='auto', execution_cache_path='', execution_excludepatterns=(), execution_timeout=30, execution_in_temp=False, execution_allow_errors=False, execution_raise_on_error=True, execution_show_tb=True, merge_streams=False, render_plugin='default', remove_code_source=False, remove_code_outputs=False, code_prompt_show='Show code cell {type}', code_prompt_hide='Hide code cell {type}', number_source_lines=False, output_stderr='show', render_text_lexer='myst-ansi', render_error_lexer='ipythontb', render_image_options={}, render_figure_options={}, render_markdown_format='myst', output_folder='build', append_css=True, metadata_to_fm=False)
Using jupyter-cache at: /home/docs/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/_readthedocs/.jupyter_cache
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 62 source files that are out of date
updating environment: [new config] 62 added, 0 changed, 0 removed
reading sources... [  2%] changelog
reading sources... [  3%] coffee_machine
reading sources... [  5%] coffee_machine_concept
/home/docs/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/docs/coffee_machine_concept.md: Skipped execution in 'auto' mode (all outputs present) [mystnb]
reading sources... [  6%] coffee_machine_logical
/home/docs/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/docs/coffee_machine_logical.md: Skipped execution in 'auto' mode (all outputs present) [mystnb]
reading sources... [  8%] coffee_machine_wrapup
reading sources... [ 10%] connect
reading sources... [ 11%] container
reading sources... [ 13%] contribute
reading sources... [ 15%] design_principles
reading sources... [ 16%] event_system
reading sources... [ 18%] first_model
reading sources... [ 19%] framework
reading sources... [ 21%] gaphor-services
/home/docs/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/docs/gaphor-services.md: Executing notebook using local CWD [mystnb]

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/.asdf/installs/python/3.12.7/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from gaphor.core.modeling import Diagram, StyleSheet
from gaphor.diagram.drop import drop

element_factory.create(StyleSheet)
diagram = element_factory.create(Diagram)

for element in element_factory.lselect():
    drop(element, diagram, x=0, y=0)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mAssertionError[0m                            Traceback (most recent call last)
Cell [0;32mIn[6], line 8[0m
[1;32m      5[0m diagram [38;5;241m=[39m element_factory[38;5;241m.[39mcreate(Diagram)
[1;32m      7[0m [38;5;28;01mfor[39;00m element [38;5;129;01min[39;00m element_factory[38;5;241m.[39mlselect():
[0;32m----> 8[0m     [43mdrop[49m[43m([49m[43melement[49m[43m,[49m[43m [49m[43mdiagram[49m[43m,[49m[43m [49m[43mx[49m[38;5;241;43m=[39;49m[38;5;241;43m0[39;49m[43m,[49m[43m [49m[43my[49m[38;5;241;43m=[39;49m[38;5;241;43m0[39;49m[43m)[49m

File [0;32m~/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/generic/multidispatch.py:134[0m, in [0;36mFunctionDispatcher.__call__[0;34m(self, *args, **kwargs)[0m
[1;32m    132[0m     logger[38;5;241m.[39mdebug([38;5;28mself[39m[38;5;241m.[39mregistry[38;5;241m.[39m_tree)
[1;32m    133[0m     [38;5;28;01mraise[39;00m [38;5;167;01mTypeError[39;00m([38;5;124mf[39m[38;5;124m"[39m[38;5;124mNo available rule found for [39m[38;5;132;01m{[39;00mtrimmed_args[38;5;132;01m!r}[39;00m[38;5;124m"[39m)
[0;32m--> 134[0m [38;5;28;01mreturn[39;00m [43mrule[49m[43m([49m[38;5;241;43m*[39;49m[43margs[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m

File [0;32m~/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/gaphor/UML/drop.py:93[0m, in [0;36mdrop_relationship_on_diagram[0;34m(element, diagram, x, y)[0m
[1;32m     91[0m [38;5;28;01mfor[39;00m head [38;5;129;01min[39;00m metadata[[38;5;124m"[39m[38;5;124mhead[39m[38;5;124m"[39m][38;5;241m.[39mget(element):
[1;32m     92[0m     [38;5;28;01mfor[39;00m tail [38;5;129;01min[39;00m metadata[[38;5;124m"[39m[38;5;124mtail[39m[38;5;124m"[39m][38;5;241m.[39mget(element):
[0;32m---> 93[0m         new_item [38;5;241m=[39m [43mdrop_relationship[49m[43m([49m[43melement[49m[43m,[49m[43m [49m[43mhead[49m[43m,[49m[43m [49m[43mtail[49m[43m,[49m[43m [49m[43mdiagram[49m[43m,[49m[43m [49m[43mx[49m[43m,[49m[43m [49m[43my[49m[43m)[49m
[1;32m     94[0m         [38;5;28;01mif[39;00m new_item:
[1;32m     95[0m             added_items[38;5;241m.[39mappend(new_item)

File [0;32m~/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/gaphor/diagram/drop.py:106[0m, in [0;36mdrop_relationship[0;34m(element, head_element, tail_element, diagram, x, y)[0m
[1;32m    104[0m     connect(item, item[38;5;241m.[39mhead, head_item)
[1;32m    105[0m [38;5;28;01mif[39;00m tail_item:
[0;32m--> 106[0m     [43mconnect[49m[43m([49m[43mitem[49m[43m,[49m[43m [49m[43mitem[49m[38;5;241;43m.[39;49m[43mtail[49m[43m,[49m[43m [49m[43mtail_item[49m[43m)[49m
[1;32m    108[0m [38;5;28;01mreturn[39;00m item

File [0;32m~/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/gaphor/diagram/presentation.py:49[0m, in [0;36mconnect[0;34m(item, handle, target)[0m
[1;32m     47[0m connector [38;5;241m=[39m ConnectorAspect(item, handle, item[38;5;241m.[39mdiagram[38;5;241m.[39mconnections)
[1;32m     48[0m sink [38;5;241m=[39m ConnectionSink(target, distance[38;5;241m=[39m[38;5;28mfloat[39m([38;5;124m"[39m[38;5;124minf[39m[38;5;124m"[39m))
[0;32m---> 49[0m [43mconnector[49m[38;5;241;43m.[39;49m[43mconnect[49m[43m([49m[43msink[49m[43m)[49m

File [0;32m~/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/gaphor/diagram/_connector.py:58[0m, in [0;36mPresentationConnector.connect[0;34m(self, sink)[0m
[1;32m     55[0m     [38;5;28;01mreturn[39;00m
[1;32m     57[0m [38;5;28mself[39m[38;5;241m.[39mconnect_handle(sink)
[0;32m---> 58[0m [43madapter[49m[38;5;241;43m.[39;49m[43mconnect[49m[43m([49m[43mhandle[49m[43m,[49m[43m [49m[43msink[49m[38;5;241;43m.[39;49m[43mport[49m[43m)[49m

File [0;32m~/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/gaphor/diagram/connectors.py:313[0m, in [0;36mRelationshipConnect.connect[0;34m(self, handle, port)[0m
[1;32m    311[0m opposite [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mline[38;5;241m.[39mopposite(handle)
[1;32m    312[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39mget_connected(opposite):
[0;32m--> 313[0m     [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mconnect_subject[49m[43m([49m[43mhandle[49m[43m)[49m
[1;32m    314[0m     line [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mline
[1;32m    315[0m     [38;5;28;01mif[39;00m line[38;5;241m.[39msubject:

File [0;32m~/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/gaphor/UML/classes/classconnect.py:35[0m, in [0;36mDependencyConnect.connect_subject[0;34m(self, handle)[0m
[1;32m     32[0m [38;5;28mself[39m[38;5;241m.[39mline[38;5;241m.[39msubject [38;5;241m=[39m relation
[1;32m     34[0m [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m relation[38;5;241m.[39mowningPackage:
[0;32m---> 35[0m     [38;5;28;01massert[39;00m [38;5;28misinstance[39m([38;5;28mself[39m[38;5;241m.[39mdiagram, UML[38;5;241m.[39mDiagram)
[1;32m     37[0m     relation[38;5;241m.[39mowningPackage [38;5;241m=[39m owner_package([38;5;28mself[39m[38;5;241m.[39mdiagram[38;5;241m.[39mowner)

[0;31mAssertionError[0m: 


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/sphinx/cmd/build.py", line 514, in build_main
    app.build(args.force_all, args.filenames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/sphinx/application.py", line 381, in build
    self.builder.build_update()
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/sphinx/builders/__init__.py", line 358, in build_update
    self.build(
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/sphinx/builders/__init__.py", line 385, in build
    updated_docnames = set(self.read())
                           ^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/sphinx/builders/__init__.py", line 502, in read
    self._read_serial(docnames)
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/sphinx/builders/__init__.py", line 567, in _read_serial
    self.read_doc(docname)
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/sphinx/builders/__init__.py", line 630, in read_doc
    publisher.publish()
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/docutils/core.py", line 234, in publish
    self.document = self.reader.read(self.source, self.parser,
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/sphinx/io.py", line 106, in read
    self.parse()
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/docutils/readers/__init__.py", line 76, in parse
    self.parser.parse(self.input, document)
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/myst_nb/sphinx_.py", line 152, in parse
    with create_client(
         ^^^^^^^^^^^^^^
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/myst_nb/core/execute/base.py", line 79, in __enter__
    self.start_client()
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/myst_nb/core/execute/direct.py", line 50, in start_client
    raise ExecutionError(str(self.path)) from result.err
myst_nb.core.execute.base.ExecutionError: /home/docs/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/docs/gaphor-services.md

Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/gaphor/envs/3621/lib/python3.12/site-packages/myst_nb/core/execute/direct.py", line 50, in start_client
    raise ExecutionError(str(self.path)) from result.err
myst_nb.core.execute.base.ExecutionError: /home/docs/checkouts/readthedocs.org/user_builds/gaphor/checkouts/3621/docs/gaphor-services.md
```

### Expected behavior
Build passes

### OS
- [X] Linux (Please put in notes the specific distro)
- [ ] macOS
- [ ] Windows

### Version
Version of Gaphor: latest main

**Comment by amolenaar** [comment]
It would be a lot easier if Jupyter did not color the stack traces 😅

**Comment by amolenaar** [comment]
Fixed by #3624.
